server/src/main/java/io/druid/curator/discovery/DiscoveryModule.java

```diff
@@ -146,6 +150,19 @@ public void configure(Binder binder)
     binder.bind(ServiceAnnouncer.class)
           .to(Key.get(CuratorServiceAnnouncer.class, Names.named(NAME)))
           .in(LazySingleton.class);
+
+    // internal discovery bindings.
+    DruidDiscoveryModule.createBindingChoices(binder, CURATOR_KEY);
```

Comment:
Why create this new method in this new class just to call it from here?  Seems like we should be able to just setup the PolyBinds directly here.